guillotina/db/storages/pg.py

```diff
@@ -568,6 +575,18 @@ def lock(self):
 
         await self.initialize_tid_statements()
 
+    def _get_index_name(self, table_name):
+        if '.' in table_name:
+            _, table_name = table_name.split(".", 1)
+
+        if table_name in ("objects", "blobs"):
+            table_name = table_name[:-1]  # singluar index name
```

Comment:
Only for statements above ^^ that I commented on but not most!

Unless `_get_index_name` is only used for statement init?

Reply:
yes, `_get_index_name` is only used in the statement init

Reply:
I just moved the code inside a function to avoid duplicating the same logic (convert table_name to singular and append schema to index_name).

All those changes shouldn't affect the current behavior.